console/console-init/ui/src/modules/project-detail/components/AccessCredentials/AccessCredentials.tsx

```diff
@@ -69,10 +65,14 @@ const AccessCredentials: React.FunctionComponent<IAccessCredentialsProps> = ({
   const onToggle = () => {
     setIsHidden(!isHidden);
   };
-  const EditIcon = isHidden ? (
-    <ToggleOffIcon onClick={onToggle} size="lg" />
-  ) : (
-    <ToggleOnIcon onClick={onToggle} size="lg" />
+  const EditIcon = (
+    <Switch
```

Comment:
Can you try to create generic/custom  component SwitchWithToggle so that we can avoid to manage state everywhere.
And replace existing switch code with custom component SwitchWithToggle.